File: python/08_069.py
def canCreateBlend(acidity, targetAcidity):

    left, right = 0, len(acidity) - 1
    
    while left < right:
        blend_acidity = (acidity[left] + acidity[right]) / 2
        
        # Account for floating-point precision
        if abs(blend_acidity - targetAcidity) < 0.001:
            return True
        elif blend_acidity < targetAcidity:
            left += 1
        else:
            right -= 1
    
    return False

File: data structures and algorithms/cs/08_069.cs
using System;

public class Solution
{
   public bool CanCreateBlend(
      double[] acidity,
      double targetAcidity)
   {
      int left = 0;
      int right = acidity.Length - 1;

      while (left < right)
      {
         double blendAcidity =
            (acidity[left] + acidity[right]) / 2;

         if (Math.Abs(blendAcidity - targetAcidity) < 0.001)
         {
            return true;
         }
         else if (blendAcidity < targetAcidity)
         {
            left++;
         }
         else
         {
            right--;
         }
      }

      return false;
   }
}
